3ds2-example/README.md

```diff
@@ -0,0 +1,91 @@
+## Adyen [3D Secure 2 Authentication](https://docs.adyen.com/online-payments/3d-secure/) Integration Example
+
+## Description
+This repository shows  the [Native](https://docs.adyen.com/online-payments/3d-secure/native-3ds2/) and [Redirect](https://docs.adyen.com/online-payments/3d-secure/redirect-3ds2/) flows for integrating 3DS2 payments using [Adyen.Drop-in and Components](https://github.com/Adyen/adyen-web).
+We will go over the integration with the [sessions flow](https://docs.adyen.com/online-payments/build-your-integration/#create-payment-session) and the [advanced flow](https://docs.adyen.com/online-payments/build-your-integration/additional-use-cases/advanced-flow-integration/).
+
+
+## Prerequisites
+- Node.js 18+ 
+
+This demo leverages Adyen's API Library for Node.js ([GitHub](https://github.com/Adyen/adyen-node-api-library) | [Docs](https://docs.adyen.com/development-resources/libraries#javascript)).
+
+
+## 1. Installation
+
+1. Clone this repo:
+
+```
+git clone https://github.com/adyen-examples/adyen-node-online-payments.git
+```
+
+
+2. Build the frontend and backend. Navigate to the /3ds2-example and install dependencies:
+
+The `frontend` directory is our client app. This is bootstrapped by [Vite](https://vitejs.dev/): a basic HTML web app with vanilla JavaScript. 
+
+The `backend` directory is our backend app. This is bootstrapped by [NestJS](https://nestjs.com/) and is written in TypeScript. This handles the native or redirect 3DS2, see services:
+- `payments.service.ts` - This uses axios to call the API.
+- `paymentsWithCheckout.service.ts` - This uses the [adyen-api node library](https://www.npmjs.com/package/@adyen/api-library) to call the API.
+> Note: The app is initiated using the `paymentsWithCheckout.service.ts` (recommended) but if you rather not use the [adyen-api node library](https://www.npmjs.com/package/@adyen/api-library) to swap out you just need to update the import for the service in the `payments.controller.ts` and `payments.module.ts` files to point to `payments.service.ts`.
+
+```
+cd 3ds2-example/frontend && npm install
+cd 3ds2-example/backend && npm install
+```
+
+## 2. Set the Environment Variables
+1. Create a `./.env` file in the `/frontend`-folder with all required configuration & run `npm install`.
+  - [Client Key](https://docs.adyen.com/user-management/client-side-authentication)
+  - `CLIENT_ENV` = `test`
+
+```
+VITE_CLIENT_KEY=YOUR_CLIENT_KEY_HERE
+CLIENT_ENV=test // environment to point to for adyen API // TODO: Check
+```
+
+2. Create a `./.env` file in the `/backend`-folder with all required configuration & run `npm install`.
+  - [API key](https://docs.adyen.com/user-management/how-to-get-the-api-key)
+  - [Merchant Account](https://docs.adyen.com/account/account-structure)
+  - `BASE_URL` = `https://checkout-test.adyen.com/v70`
+
+```
+API_KEY=YOUR_API_KEY
+MERCHANT_ACCOUNT=YOUR_MERCHANT_ACCOUNT
+BASE_URL=https://checkout-test.adyen.com/v70
```

Comment:
Hmm, if it's not, then let's keep the BASE_URL as is. https://github.com/Adyen/adyen-node-api-library?tab=readme-ov-file#custom-http-client-configuration

Reply:
The BASE_URL is used for the example that doesn't use Adyen node-api library, there are two BE implementations, one with the library and one without. I thought it would be helpful but now realize it might lead to more confusion so maybe we should just stick to the example using the library? what do you think?

Reply:
I'm going to remove the non-library service so the example is more focused. 💪

Reply:
✅

Reply:
👍